plugins/@grouparoo/demo/src/events.ts

```diff
@@ -0,0 +1,309 @@
+import { runAction } from "./util/runAction";
+import { ApiKey, App, ProfilePropertyRule } from "@grouparoo/core";
+import { getPurchaseCategories, getPurchases } from "./sample_data";
+import * as uuid from "uuid";
+import { log, userCreatedAt } from "./util/shared";
+
+export async function events() {
+  await enableEventsApp();
+
+  const apiKey = await getApiKey();
+  const categories = await getPurchaseCategories();
+  const purchases = await getPurchases();
+  await generateBrowseEvents(purchases.length * 1.25, apiKey, categories);
+  await generatePurchaseEvents(purchases, apiKey);
+}
+
+async function generateBrowseEvents(count, apiKey, categories) {
+  for (let i = 0; i < count; i++) {
+    const session = new MockSession(`browse${i}`, apiKey, null, categories);
+    await session.run();
+  }
+}
+
+async function generatePurchaseEvents(purchases, apiKey) {
+  for (const purchase of purchases) {
+    const session = new MockSession(
+      `purchase${purchase.id}`,
+      apiKey,
+      purchase,
+      []
+    );
+    await session.run();
+  }
+}
+
+// copied from initializer/events.ts
+async function addEventsApp() {
+  let eventsApp = await App.scope(null).findOne({
+    where: {
+      type: "events",
+    },
+  });
+  if (!eventsApp) {
+    eventsApp = App.build({
+      type: "events",
+      name: "events",
+      state: "draft",
+    });
```

Comment:
In what cases is the event app not being built?  I guess if you have a totally blank database and start with the `cli` commands the start steps won't run...

If you want to share that code, you could add an `export` in front of addEventsApp() in the core initializer and you can require it thought he node module, something like 
`import {addEventsApp} from "@gruoparoo/core/dist/initializers/events"`

Reply:
Yes, that was the case.
CLI doesn't call those initalizers in that way.
I think it's ok for now.